extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/metrics/HibernateCounter.java

```diff
@@ -0,0 +1,141 @@
+package io.quarkus.hibernate.orm.runtime.metrics;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.stat.Statistics;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.hibernate.orm.runtime.JPAConfig;
+
+public class HibernateCounter implements org.eclipse.microprofile.metrics.Counter {
+
+    private volatile SessionFactory sessionFactory;
+    private String persistenceUnitName;
+    private String metric;
+
+    public HibernateCounter() {
+    }
+
+    public HibernateCounter(String persistenceUnitName, String metric) {
+        this.persistenceUnitName = persistenceUnitName;
+        this.metric = metric;
+    }
+
+    public String getPersistenceUnitName() {
+        return persistenceUnitName;
+    }
+
+    public void setPersistenceUnitName(String persistenceUnitName) {
+        this.persistenceUnitName = persistenceUnitName;
+    }
+
+    public String getMetric() {
+        return metric;
+    }
+
+    public void setMetric(String metric) {
+        this.metric = metric;
+    }
+
+    @Override
+    public long getCount() {
+        Statistics metrics = getSessionFactory().getStatistics();
```

Comment:
It's very minor but it would be easier to understand if this variable was called `statistics`.